virtwho/configure.py

```diff
@@ -1,4 +1,119 @@
+        """
+        self.remote_ssh = virtwho_ssh_connect(mode)
+        self.local_file = f'/root/{mode}.conf'
+        self.remote_file = f'/etc/virt-who.d/{mode}.conf'
+        self.section = f'virtwho-{mode}'
+        self.cfg = Configure(self.local_file, self.remote_ssh, self.remote_file)
+        if mode == 'local':
+            logger.info("Don't need to configure for local mode")
+        else:
+            self.update('type', mode)
+            if mode == 'esx':
+                self.update('server', config.vcenter.server)
+                self.update('username', config.vcenter.username)
+                self.update('password', config.vcenter.password)
+            elif mode == 'xen':
+                self.update('server', config.xen.server)
+                self.update('username', config.xen.username)
+                self.update('password', config.xen.password)
+            elif mode == 'hyperv':
+                self.update('server', config.hyperv.server)
+                self.update('username', config.hyperv.username)
+                self.update('password', config.hyperv.password)
+            elif mode == 'rhevm':
+                self.update('server', config.rhevm.server)
+                self.update('username', config.rhevm.username)
+                self.update('password', config.rhevm.password)
+            elif mode == 'libvirt':
+                self.update('server', config.libvirt.server)
+                self.update('username', config.libvirt.username)
+                self.update('password', config.libvirt.password)
+            elif mode == 'kubevirt':
+                self.update('kubeconfig', config.mode.config_file)
+            if register == "rhsm":
+                self.update('owner', config.rhsm.default_org)
+            elif register == 'satellite':
+                self.update('owner', config.satellite.default_org)
+            self.update('hypervisor_id', 'hostname')
+
+    def update(self, option, value):
+        """Add or update an option
+        :param option: Option will be added if not exist.
+        :param value: Value to update for the option.
+        """
+        self.cfg.update(self.section, option, value)
+
+    def delete(self, option):
+        """Delete an option
+        :param option: Option to remove.
+        """
+        self.cfg.delete(self.section, option)
+
+    def destroy(self):
+        """Remove the /etc/virt-who.d/{mode}.conf file"""
+        self.remote_ssh.remove_file(self.remote_file)
+
+
+class VirtwhoGlobalConfig:
+    """Able to manage /etc/virt-who.conf file when call this class"""
+    def __init__(self, mode=None):
+        """
+        :param mode: Hypervisor mode. When mode is local, will manage
+            the local libvirt host, othervise will manage the host for
+            all other remote modes.
+        """
+        self.remote_ssh = virtwho_ssh_connect(mode)
+        self.local_file = '/root/virt-who.conf'
+        self.remote_file = '/etc/virt-who.conf'
```

Comment:
Check the virt-who.conf.save is exist in temp directory or not
```
        self.save_file = os.path.join(TEMP_DIR, 'virt-who.conf.save')
        if not os.path.exists(self.save_file):
            self.remote_ssh.get_file(self.remote_file, self.save_file)
```

Reply:
I also thought about if need to download the virt-who.conf file if not exists, but all contents are commented in /etc/virt-who.conf and our code will delete all comment lines when parse the ini file, so it seems that no need to download one copy, the default file just like an empty file.
What I want to change is editing the Configure() in settings.py to avoid update the default file information when just call the Configure() class but never do other things. Infect this is not necessary, because when we call the class that is because we want to update the content.
```
    def __init__(self, local_file, remote_ssh=None, remote_file=None):
        self.local_file = local_file
        self.remote_file = remote_file
        self.remote_ssh = remote_ssh
        self.config = ConfigParser(dict_type=AttrDict)
        self.config.read(self.local_file)
        for key in self.config._sections.keys():
            setattr(self, key, getattr(self.config._sections, key))

    def save(self):
        with open(self.local_file, 'w') as f:
            self.config.write(f, space_around_delimiters=False)
        if self.remote_ssh and self.remote_file:
            self.remote_ssh.put_file(self.local_file, self.remote_file)
```